BedrockCore.cpp

```diff
@@ -29,6 +29,28 @@ class AutoScopeRewrite {
     bool (*_handler)(int, const char*, string&);
 };
 
+// RAII-style mechanism for automatically setting and unsetting an on prepare handler
+class AutoScopeOnPrepare {
+  public:
+    AutoScopeOnPrepare(bool enable, SQLite& db, void (*handler)(SQLite& _db, int64_t tableID)) : _enable(enable), _db(db), _handler(handler) {
+        if (_enable) {
+            _db.setOnPrepareHandler(_handler);
+            _db.enablePrepareNotifications(true);
```

Comment:
I tried that and it didn't work, so I just copied the implementation of the rewrite handler.

Reply:
I think it is because of the way we pass the pointer handler around, it ends up pointing to a valid reference to a null pointer. i didn't spend a lot of time on it.

Reply:
ok all good especially if it is the same as the rewrite handler